WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderSearchSuggestionRecyclerAdapter.java

```diff
@@ -0,0 +1,165 @@
+        final int count = mCursor == null ? 0 : mCursor.getCount();
+        return count > 0 ? count + 1 : 0; // we add an extra row at the end to show the "Clear search history" button
+    }
+
+    @Override
+    public long getItemId(int position) {
+        if (isLast(position)) {
+            return CLEAR_ALL_ROW_ID;
+        } else if (!mCursor.moveToPosition(position)) {
+            throw new IllegalStateException("couldn't move cursor to position " + position);
+        }
+        return mCursor.getLong(mCursor.getColumnIndex(ReaderSearchTable.COL_ID));
+    }
+
+    private boolean isLast(final int position) {
+        return position == mCursor.getCount();
+    }
+
+    public void swapCursor(final Cursor newCursor) {
+        if (newCursor == mCursor) return;
+        SqlUtils.closeCursor(mCursor);
+        mCursor = newCursor;
+        notifyDataSetChanged();
+    }
+
+    public void setOnSuggestionClickListener(OnSuggestionClickListener onSuggestionClickListener) {
+        mOnSuggestionClickListener = onSuggestionClickListener;
+    }
+
+    public void setOnSuggestionDeleteClickListener(OnSuggestionDeleteClickListener onSuggestionDeleteClickListener) {
+        mOnSuggestionDeleteClickListener = onSuggestionDeleteClickListener;
+    }
+
+    public void setOnSuggestionClearClickListener(OnSuggestionClearClickListener onSuggestionClearClickListener) {
+        mOnSuggestionClearClickListener = onSuggestionClearClickListener;
+    }
+
+    public void reload() {
+        setQuery(mCurrentQuery, true);
+    }
+
+    public void setQuery(final String newQuery) {
+        setQuery(newQuery, false);
+    }
+
+    private void setQuery(final String newQuery, final boolean forceUpdate) {
+        if (!forceUpdate && newQuery != null && newQuery.equalsIgnoreCase(mCurrentQuery) && mCursor != null) {
+            return;
+        }
+        mCurrentQuery = newQuery;
+        final int maxSuggestions = newQuery == null ? MAX_SUGGESTIONS_WHEN_EMPTY : MAX_SUGGESTIONS;
+        swapCursor(ReaderSearchTable.getQueryStringCursor(newQuery, maxSuggestions));
+    }
+
+    class SearchSuggestionHolder extends RecyclerView.ViewHolder {
+        private final ImageView mHistoryImageView;
+        private final TextView mSuggestionTextView;
+        private final ImageView mDeleteImageView;
+
+        SearchSuggestionHolder(final View itemView) {
+            super(itemView);
+            mHistoryImageView = itemView.findViewById(R.id.image_history);
+            mSuggestionTextView = itemView.findViewById(R.id.text_suggestion);
+            mDeleteImageView = itemView.findViewById(R.id.image_delete);
+        }
+    }
+
+    public interface OnSuggestionClickListener {
+        void onSuggestionClicked(String query);
+    }
+
+    public interface OnSuggestionDeleteClickListener {
+        void onDeleteClicked(String query);
+    }
+
+    public interface OnSuggestionClearClickListener {
+        void onClearClicked();
+    }
```

Comment:
Maybe we can consider to move this interfaces into a dedicated kotlin file. In this way we could avoid to duplicate some of them in the `ReaderSearchSuggestionAdapter`; wdyt?

NOTE: usually we try to create new classes (like this one) in kotlin where possible; but this is just to share a general comment (so no action required actually 😊).

Reply:
Thanks for the review! 😊 I have now extracted those interfaces to a single Kotlin file.